File: dacars-backend/src/main/java/com/dacars/dacars_backend/util/BildHandler.java
package com.dacars.dacars_backend.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.util.StringUtils;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardCopyOption;

public class BildHandler {

    private static final Logger log = LoggerFactory.getLogger(BildHandler.class);

    private static final String BILDER_ORDNER = "src/main/resources/static/autobilder";

    static {
        try {
            Path ordnerPath = Paths.get(BILDER_ORDNER);
            if (!Files.exists(ordnerPath)) {
                Files.createDirectories(ordnerPath);
            }
        } catch (IOException e) {
            throw new RuntimeException("Fehler beim Erstellen des Bilderverzeichnisses", e);
        }
    }

    public static String speichereBild(MultipartFile file, long autoId, int index) {
        String originalDateiname = StringUtils.cleanPath(file.getOriginalFilename());
        String einzigartigerDateiname = autoId + "_" + index + "_" + originalDateiname;

        try {
            if (file.isEmpty()) {
                throw new IOException("Die Datei ist leer: " + originalDateiname);
            }

            Path zielPfad = Paths.get(BILDER_ORDNER).resolve(einzigartigerDateiname);
            Files.copy(file.getInputStream(), zielPfad, StandardCopyOption.REPLACE_EXISTING);

            return einzigartigerDateiname;
        } catch (IOException e) {
            throw new RuntimeException("Fehler beim Speichern der Datei: " + originalDateiname, e);
        }
    }

    public static void loescheBild(String dateiname) {
        log.info(" ===== loescheBild ENTER dateiname={}", dateiname);
        Path bildPfad = Paths.get(BILDER_ORDNER).resolve(dateiname);
        if (Files.exists(bildPfad)) {
            try {
                Files.delete(bildPfad);
                log.info("Bild erfolgreich gelöscht: {}", dateiname);
            } catch (IOException e) {
                log.error("Fehler beim Löschen des Bildes: {}", dateiname, e);
            }
        } else {
            log.warn("Bild nicht gefunden: {}", dateiname);
        }
    }
    
}

File: backend-nodejs/utils/bildHandler.js
const fs = require('fs').promises;
const path = require('path');

class BildHandler {
    constructor() {
        this.BILDER_ORDNER = process.env.UPLOAD_DIR || './uploads/bilder';
        this.init();
    }

    async init() {
        try {
            await fs.mkdir(this.BILDER_ORDNER, { recursive: true });
            console.log(`📁 Upload-Verzeichnis erstellt: ${this.BILDER_ORDNER}`);
        } catch (error) {
            console.error('❌ Fehler beim Erstellen des Bilderverzeichnisses:', error.message);
        }
    }

    async speichereBild(file, autoId, index) {
        try {
            if (!file || !file.buffer) {
                throw new Error('Ungültige Datei');
            }

            const originalDateiname = file.originalname || `bild_${Date.now()}.jpg`;
            const extension = path.extname(originalDateiname) || '.jpg';
            const einzigartigerDateiname = `${autoId}_${index}_${Date.now()}${extension}`;
            
            const zielPfad = path.join(this.BILDER_ORDNER, einzigartigerDateiname);
            await fs.writeFile(zielPfad, file.buffer);
            
            console.log(`✅ Bild gespeichert: ${einzigartigerDateiname}`);
            return einzigartigerDateiname;
        } catch (error) {
            console.error('❌ Fehler beim Speichern des Bildes:', error.message);
            throw error;
        }
    }

    async loescheBild(dateiname) {
        console.log(`🗑️  Lösche Bild: ${dateiname}`);
        const bildPfad = path.join(this.BILDER_ORDNER, dateiname);
        
        try {
            await fs.access(bildPfad);
            await fs.unlink(bildPfad);
            console.log(`✅ Bild gelöscht: ${dateiname}`);
        } catch (error) {
            if (error.code === 'ENOENT') {
                console.warn(`⚠️  Bild nicht gefunden: ${dateiname}`);
            } else {
                console.error(`❌ Fehler beim Löschen des Bildes ${dateiname}:`, error.message);
            }
        }
    }
}

module.exports = new BildHandler();